src/dmqnode/storage/Ring.d

```diff
@@ -302,15 +302,19 @@ public class RingNode : StorageChannels
 
         ***********************************************************************/
 
-        override protected void push_ ( char[] value )
+        override protected void push_ ( cstring value )
         {
             this.outer.dmqnode.record_action_counters.increment("pushed", value.length);
 
             // For the temporary console log only:
             this.records_pushed++;
             this.bytes_pushed += value.length;
 
-            if (!this.queue.push(cast(ubyte[])value))
+            if (auto dst = this.queue.push(value.length))
+            {
+                dst[] = cast(Const!(ubyte)[])value;
```

Comment:
Please, try to make yourself a habit - each and every single time you find yourself writing `cast`, spend few minutes more trying to express same semantics without a cast ;)

Reply:
In this case for example:

```D
void[] dst_raw = dst;
Const!(void)[] value_raw = value;
dst_raw[] = value_raw[];
```